### DISCUSSION TITLE
Recovering Lima instance

### DISCUSSION THREAD

**Discussion by steveafrost** [lima-vm/lima#2288] [Q&A] [answered]
Well, I've gone and done it. That is, ran `limactl delete {instance}` in haste and removed a VM that had data that I still needed. Is there any way to undo a delete or restore a previous VM instance? I hadn't setup snapshots yet.

Thanks for your help or any pointing in a direction where I may be able to explore any/all possibilities.

**Comment by jandubois** [comment, marked as answer]
I'm afraid it is gone unless you have some kind of backup.

If you are on macOS, check Time Machine. It may keep hourly APFS snapshots on the local disk, if it has enough free space. But you must hurry, they can expire quickly.

**Reply by jandubois** [reply]
Just to check which snapshots are available:

```console
$ tmutil listlocalsnapshotdates
Snapshot dates for all disks:
2024-04-18-144128
2024-04-18-154135
…
2024-04-19-104505
2024-04-19-114515
2024-04-19-124523
2024-04-19-134539
```

**Reply by steveafrost** [reply]
Ufda! Yup no snapshots. Thanks for the tip and I'll keep it in mind for next time to make more backups. I do it with all my other data but got too courageous this time. 😆
